exchanges/poloniex/poloniex_types.go

```diff
@@ -446,75 +125,906 @@ var WithdrawalFees = map[currency.Code]float64{
+}
+
+// Withdraw holds withdraw information
+type Withdraw struct {
+	WithdrawRequestID uint64 `json:"withdrawalRequestsId"`
+}
+
+// WithdrawCurrencyRequest represents a V2 currency withdrawal parameter.
+type WithdrawCurrencyRequest struct {
+	Coin        currency.Code `json:"coin"`
+	Network     string        `json:"network"`
+	Amount      float64       `json:"amount,string"`
+	Address     string        `json:"address"`
+	AddressTag  string        `json:"addressTag,omitempty"`
+	AllowBorrow bool          `json:"allowBorrow,omitempty"`
+}
+
+// AccountMargin represents an account margin response
+type AccountMargin struct {
+	TotalAccountValue types.Number `json:"totalAccountValue"`
+	TotalMargin       types.Number `json:"totalMargin"`
+	UsedMargin        types.Number `json:"usedMargin"`
+	FreeMargin        types.Number `json:"freeMargin"`
+	MaintenanceMargin types.Number `json:"maintenanceMargin"`
+	CreationTime      types.Time   `json:"time"`
+	MarginRatio       string       `json:"marginRatio"`
+}
+
+// BorrowStatus represents currency borrow status.
+type BorrowStatus struct {
+	Currency         string       `json:"currency"`
+	Available        types.Number `json:"available"`
+	Borrowed         types.Number `json:"borrowed"`
+	Hold             types.Number `json:"hold"`
+	MaxAvailable     types.Number `json:"maxAvailable"`
+	HourlyBorrowRate types.Number `json:"hourlyBorrowRate"`
+	Version          string       `json:"version"`
+}
+
+// MaxBuySellAmount represents a maximum buy and sell amount.
+type MaxBuySellAmount struct {
+	Symbol           string       `json:"symbol"`
+	MaxLeverage      int64        `json:"maxLeverage"`
+	AvailableBuy     types.Number `json:"availableBuy"`
+	MaxAvailableBuy  types.Number `json:"maxAvailableBuy"`
+	AvailableSell    types.Number `json:"availableSell"`
+	MaxAvailableSell types.Number `json:"maxAvailableSell"`
+}
+
+// PlaceOrderRequest represents place order parameters.
+type PlaceOrderRequest struct {
+	Symbol      currency.Pair `json:"symbol"`
+	Side        string        `json:"side"`
+	Type        orderType     `json:"type,omitempty"`
+	AccountType string        `json:"accountType,omitempty"`
+
+	// Quantity Base units for the order. Quantity is required for MARKET SELL or any LIMIT orders
+	Quantity float64 `json:"quantity,omitempty,string"`
+
+	// Amount Quote units for the order. Amount is required for MARKET BUY order
+	Amount float64 `json:"amount,omitempty,string"`
+
+	// Price is required for non-market orders
+	Price float64 `json:"price,omitempty,string"`
+
+	TimeInForce   timeInForce `json:"timeInForce,omitempty"` // GTC, IOC, FOK (Default: GTC)
+	ClientOrderID string      `json:"clientOrderId,omitempty"`
+
+	AllowBorrow bool   `json:"allowBorrow,omitempty"`
+	STPMode     string `json:"stpMode,omitempty"` // self-trade prevention. Defaults to EXPIRE_TAKER. None: enable self-trade; EXPIRE_TAKER: Taker order will be canceled when self-trade happens
```

Comment:
🚧 Change request:
* Inconsistency; This is named differently in futures
I'll cover this in the patch.
```suggestion
	 SelfTradePreventionMode:     string `json:"stpMode,omitempty"` // self-trade prevention. Defaults to EXPIRE_TAKER. None: enable self-trade; EXPIRE_TAKER: Taker order will be canceled when self-trade happens
```